src/Sitecore.Modules.WeBlog/Managers/EntryManager.cs

```diff
@@ -254,41 +254,49 @@ public virtual EntryItem[] GetBlogEntriesByMonthAndYear(Item blog, int month, in
         /// <returns>An array of EntryItem classes</returns>
         public virtual EntryItem[] GetPopularEntriesByView(Item blogItem, int maxCount)
         {
-            var blogEntries = GetBlogEntries(blogItem);
-            var entryIds = (from entry in blogEntries select entry.ID).ToArray();
-
-            if (entryIds.Any())
+            if (AnalyticsEnabled())
             {
-                var views = new Dictionary<ID, long>();
+                var blogEntries = GetBlogEntries(blogItem);
+                var entryIds = (from entry in blogEntries select entry.ID).ToArray();
 
-                foreach (var id in entryIds)
+                if (entryIds.Any())
                 {
-#if FEATURE_XDB
-                    var query = new ItemVisitsQuery(this.ReportDataProvider)
+                    var views = new Dictionary<ID, long>();
+
+                    foreach (var id in entryIds)
                     {
-                        ItemId = id
-                    };
+#if FEATURE_XDB
+                        var query = new ItemVisitsQuery(this.ReportDataProvider)
+                        {
+                            ItemId = id
+                        };
 
-                    query.Execute();
+                        query.Execute();
 
-                    var itemViews = query.Visits;
+                        var itemViews = query.Visits;
 #elif FEATURE_DMS
                     var queryId = id.ToString().Replace("{", string.Empty).Replace("}", string.Empty);
                     var sql = "SELECT COUNT(ItemId) as Visits FROM {{0}}Pages{{1}} WHERE {{0}}ItemId{{1}} = '{0}'".FormatWith(queryId);
 
                     var itemViews = DataAdapterManager.ReportingSql.ReadOne(sql, reader => DataAdapterManager.ReportingSql.GetLong(0, reader));
 #endif
 
-                    if (itemViews > 0)
-                        views.Add(id, itemViews);
-                }
+                        if (itemViews > 0)
+                            views.Add(id, itemViews);
+                    }
 
-                var ids = views.OrderByDescending(x => x.Value).Take(maxCount).Select(x => x.Key).ToArray();
+                    var ids = views.OrderByDescending(x => x.Value).Take(maxCount).Select(x => x.Key).ToArray();
 
-                return (from id in ids select blogEntries.First(i => i.ID == id)).ToArray();
+                    return (from id in ids select blogEntries.First(i => i.ID == id)).ToArray();
+                }
             }
-            else
-                return new EntryItem[0];
+            return new EntryItem[0];
+        }
```

Comment:
Could you put an "else" here and write a log message that analytics is disabled, so this component won't render anything? It will help when people try and use the component without analytics and wonder why it's not working.